metaflow/plugins/pypi/micromamba.py

```diff
@@ -91,32 +108,40 @@ def solve(self, id_, packages, python, platform):
                 cmd.append("python==%s" % python)
             # TODO: Ensure a human readable message is returned when the environment
             #       can't be resolved for any and all reasons.
-            return [
+            solve = [
                 {k: v for k, v in item.items() if k in ["url"]}
                 for item in self._call(cmd, env)["actions"]["LINK"]
             ]
+            self.logger(
+                f"Solved {platform} virtual environment {id_} in {time.time() - start_time:.2f}s!"
+            )
+            return solve
 
     def download(self, id_, packages, python, platform):
         # Unfortunately all the packages need to be catalogued in package cache
         # because of which this function can't be parallelized
-
-        # Micromamba is painfully slow in determining if many packages are infact
-        # already cached. As a perf heuristic, we check if the environment already
-        # exists to short circuit package downloads.
-        if self.path_to_environment(id_, platform):
-            return
-
         prefix = "{env_dirs}/{keyword}/{platform}/{id}".format(
             env_dirs=self.info()["envs_dirs"][0],
             platform=platform,
             keyword="metaflow",  # indicates metaflow generated environment
             id=id_,
         )
 
-        # Another forced perf heuristic to skip cross-platform downloads.
+        # Micromamba is painfully slow in determining if many packages are infact
+        # already cached. As a perf heuristic, we check if the environment already
+        # exists to short circuit package downloads.
+
+        # cheap check
         if os.path.exists(f"{prefix}/fake.done"):
             return
 
+        # somewhat expensive check
+        # TODO: make this less expensive
+        if self.path_to_environment(id_, platform):
```

Comment:
during forming a new environment, would this not cache the path to environment as `None` due to it not being ready yet?